spec/lib/carma/client/mule_soft_auth_token_client_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'carma/client/mule_soft_auth_token_client'
+
+describe CARMA::Client::MuleSoftAuthTokenClient do
+  let(:client) { described_class.new }
+
+  let(:config) { double('config') }
+  let(:timeout) { 60 }
+  let(:settings) do
+    OpenStruct.new(
+      token_url: 'my/token/url',
+      client_id: 'id',
+      client_secret: 'secret',
+      timeout:
+    )
+  end
+
+  before do
+    allow(client).to receive(:config).and_return(config)
+    allow(config).to receive_messages(timeout:, settings:)
+  end
+
+  describe '#new_bearer_token' do
+    subject { client.new_bearer_token }
+
+    let(:token_url) { 'oauth2/ause1x1h6Zit9ziQL0j6/v1/token' }
```

Comment:
🤔 Would it make sense to DRY this out to a constant in the client and reference it from here?

Reply:
I think that's a great idea 👍